include/meta/corpus/corpus.h

```diff
@@ -75,6 +75,16 @@ class corpus
     virtual document next() = 0;
 
     /**
+     * Reset the corpus to the beginning.
+     */
+    virtual void reset() = 0;
+
+    /**
+     * Skip the next n documents in this corpus (default = 1).
+     */
+    virtual void skip(uint64_t n = 1) = 0;
```

Comment:
Since this was our quick and hacky (and slow) way of parallelizing the corpus before, I think you can get rid of it and all the implementations in the various subclasses.